ompi/communicator/comm_cid.c

```diff
@@ -858,16 +859,36 @@ static int ompi_comm_allreduce_pmix_reduce_complete (ompi_comm_request_t *reques
     opal_dss.unload(&sbuf, (void**)&info.data.bo.bytes, &info.data.bo.size);
     OBJ_DESTRUCT(&sbuf);
 
-    if (cid_context->send_first) {
-        (void)asprintf(&info.key, "%s:%s:send:%d", cid_context->port_string, cid_context->pmix_tag,
-                       cid_context->iter);
-        (void)asprintf(&pdat.value.key, "%s:%s:recv:%d", cid_context->port_string, cid_context->pmix_tag,
-                       cid_context->iter);
+    bytesWritten = asprintf(&info.key,
+                            cid_context->send_first ? "%s:%s:send:%d"
+                                                    : "%s:%s:recv:%d",
+                            cid_context->port_string,
+                            cid_context->pmix_tag,
+                            cid_context->iter);
+
+    if (bytesWritten == -1) {
+        fprintf (stderr, "writing info.key failed\n");
```

Comment:
We used `fprintf` elsewhere in here?  Oh, how embarrassing...  😳 

Yes, if you could have a 2nd commit on this PR that fixes all the other `fprintf`s, that would be great.  Thanks!